### PR TITLE
completed fetch in react practice

### PR DESCRIPTION
**Pull request by bryan-jun**
https://user-images.githubusercontent.com/61750728/173489938-be22dc6b-7d32-4bb1-9afd-7d0f6517ee1c.mov

### CODE REVIEW

File: fetch-in-react-practice/client/components/app.jsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  }
+
+  componentDidMount() {
+    fetch('/api/todos')
+      .then(response => response.json())
+      .then(data => {
+        const todoList = [];
+        for (let i = 0; i < data.length; i++) {
+          todoList.push(data[i]);
+        }
```
Comment (scbowler) [inline fetch-in-react-practice/client/components/app.jsx]: No need for a loop here, you can directly set `data` to the `todos` property in state with `setState`

File: fetch-in-react-practice/client/components/app.jsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(opposite)
+    }
+
+    )
+      .then(response => response.json())
+      .then(data => {
+        const todoList3 = this.state.todos;
```
Comment (scbowler) [inline fetch-in-react-practice/client/components/app.jsx]: This won't make a copy of the array, it just creates another reference to the same array. So on the next line when you make changes to the array you are mutating the state which can cause your app to not update correctly and/or corrupt your data in state. To fix this you need to make an actual copy of the array, below are 2 common ways to copy an array.

Using [slice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice)
```JAVASCRIPT
const todoList3 = this.state.todos.slice();
```

Using ES6 [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax)
```JAVASCRIPT
const todoList3 = [...this.state.todos];
```

**Review by scbowler — CHANGES_REQUESTED** [review]
Looking good, just one update needed to prevent the mutation of state. Let me know if you have any questions.

File: fetch-in-react-practice/client/components/app.jsx (hunk 2 above)
Comment (bryan-jun) [reply fetch-in-react-practice/client/components/app.jsx]: Thank you so much. Used the spread operator!

**Review by scbowler — APPROVED** [review]
Great job implementing the feedback, looks good! 👍

**Pull request merged**